sopel/tests/factories.py

```diff
@@ -67,6 +81,16 @@ def preloaded(
         return mockbot
 
     def __call__(self, settings: config.Config) -> bot.Sopel:
+        """Create a test Sopel instance.
+
+        :param settings: test settings used by the test bot
+        :return: an instance of Sopel ready for testing purpose
+
+        This method will create an instance of :class:`~sopel.bot.Sopel` using
+        the ``settings`` provided and a
+        :class:`~sopel.tests.mocks.MockIRCBackend` that can be used to fake
+        messages received from an IRC server.
```

Comment:
The mention of the backend swap-out here feels a little oblique to me. I would perhaps add a mention of the fake backend to the one-line summary.